src/main/kotlin/com/codingchallenge/transactionanalyser/TransactionProcessor.kt

```diff
@@ -0,0 +1,24 @@
+package com.codingchallenge.transactionanalyser
+
+import com.codingchallenge.transactionanalyser.domain.Transaction
+import com.codingchallenge.transactionanalyser.domain.UserInput
+import java.math.BigDecimal
+
+class TransactionProcessor {
+    fun filterTransactionsByUserInput(transactions: List<Transaction>, input: UserInput): List<Transaction>{
+        return transactions
```

Comment:
consider the use of comments here. Generally I feel comments are making your code less readable here.
Your code itself is doing many things in each line so perhaps you thought a comment will help the reader? think again and consider the person reading the code and why that might be. We can refactor our code further an avoid the comments completely. Think about how each `.filter` could be read as plain english. 

For example: let's look at line 11.
I could make a smaller private function below this class (or a Kotlin extension function even) which extracts out `it.fromAccountId == input.accountId` and we call that function `isMatchingFromAccount(...)` or something and then we extract out `it.toAccountId == input.accountId` to another small private function or a Kotlin extension function and call it `isMatchingtoAccountId(...)` or something, we then have made small functions that do one thing and one thing well. 
Your line 11 will then read as something like:
`.filter { isFromAccountMatching(...) || isToAccountMatching(...) }`
it makes it clearer for the code to express itself in plain english rather than in a comment

Reply:
Very valid point. Readable code > comments for sure.